ENH: Allow multichannel input

richford opened the pull request:
## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary
<!--- What does your code do? -->
This PR adds support for multichannel input in `get_dataset` and all necessary supporting functions. It also adds support for remote storage using `fsspec`.

I wrote this mainly for my own use but I think it might be more broadly useful upstream as well, provided we can align it with the vision of the main nobrainer devs.

## Checklist
<!--- Please, let us know if you need help-->
- [x] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)

## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

richford commented:
Resolves #176

Aakanksha-Rana commented:
Thanks @richford. I will review this. Could you please also add tests for these added functionalities?

richford commented:
Hi @Aakanksha-Rana, thanks! I added some tests for volume.py and io.py. I'll add further tests for the added functionality in  transform.py and datasets.py later (probably next week). I also added tests for the pre-existing functions in volume.py.
